### PR TITLE
Implement pending spec

### PR DESCRIPTION
**Pull request by ct-clearhaus**
Alternatively we should adjust PedicelPay to allow for this, but I think it is quite a tight corner case, so we could accept this 💩 spec or just not spec it. WDYT?

### CODE REVIEW

File: spec/lib/pedicel/base_spec.rb [hunk 1]
```diff
@@ -231,11 +231,75 @@
         is_expected.to be_truthy
       end
 
-      it 'handles that once certificate can be both intermediate and leaf'
-
       it 'extracts leaf, intermediate, and root' do
         is_expected.to eq([backend.leaf_certificate, backend.intermediate_certificate, backend.ca_certificate])
       end
+
+      it 'handles that once certificate can be both intermediate and leaf', me:true do
```
Comment (mt-clearhaus) [inline spec/lib/pedicel/base_spec.rb]: `me:true`

Comment (mt-clearhaus) [reply spec/lib/pedicel/base_spec.rb]: s/once certificate/one certificate/

**Review by mt-clearhaus — CHANGES_REQUESTED** [review]
I see no harm in having this spec. We can change PedicelPay later if the need arises and then simplify this spec.

File: spec/lib/pedicel/base_spec.rb (hunk 1 above)
Comment (ct-clearhaus) [reply spec/lib/pedicel/base_spec.rb]: 🙇

**Review by mt-clearhaus — DISMISSED** [review]

**Comment by ct-clearhaus** [comment]
It's quite mystifying why the specs now fail on Ruby 2.3.7; it seems `libssl-dev` is missing.

**Comment by mt-clearhaus** [comment]
However, it is certainly missing from 2.3.7 :confused: 

```
$ docker run --rm -it ruby:2.3.3 bash -c 'find / -type f -name evp.h'
/usr/include/openssl/evp.h
$ docker run --rm -it ruby:2.3.7 bash -c 'find / -type f -name evp.h'
# no output

$ docker run --rm -it ruby:2.3.3 bash -c 'dpkg -l | grep -o " libssl-dev"'
 libssl-dev
$ docker run --rm -it ruby:2.3.7 bash -c 'dpkg -l | grep -o " libssl-dev"'
# no output
```

**Comment by mt-clearhaus** [comment]
It we say that one cannot expect OpenSSL development libraries to be installed in a generic Ruby Docker image, a solution could be to always explicitly install them... Something like the following CircleCI config diff might work.

```
--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,20 +1,26 @@
 version: 2
 
+common_steps: &common_steps
+  - checkout
+  - run:
+      name: Prepare environment
+      command: |
+        apt-get update
+        apt-get install -y --no-install-recommends libssl-dev
+        bundle install
+  - run:
+      name: Test
+      command: bundle exec rake test
+
 jobs:
   ruby233:
     docker:
       - image: ruby:2.3.3
-    steps:
-      - checkout
-      - run: bundle install
-      - run: bundle exec rake test
+    steps: *common_steps
   ruby237:
     docker:
       - image: ruby:2.3.7
-    steps:
-      - checkout
-      - run: bundle install
-      - run: bundle exec rake test
+    steps: *common_steps
 
 workflows:
   version: 2
```

**Comment by mt-clearhaus** [comment]
It works :slightly_smiling_face: https://circleci.com/gh/clearhaus/pedicel/105

**Review by ct-clearhaus — COMMENTED** [review]
Great! 👍

**Review by mt-clearhaus — APPROVED** [review]

**Pull request merged**